tools/ci_connector_ops/ci_connector_ops/pipelines/publish.py

```diff
@@ -30,14 +29,13 @@ async def _run(self) -> StepResult:
             .with_env_variable("CACHEBUSTER", str(uuid.uuid4()))
             .with_exec(["ls", docker_repository])
         )
-        crane_ls_exit_code = await with_exit_code(crane_ls)
-        crane_ls_stderr = await with_stderr(crane_ls)
-        crane_ls_stdout = await with_stdout(crane_ls)
-        if crane_ls_exit_code != 0:
-            if "NAME_UNKNOWN" in crane_ls_stderr:
+        try:
+            crane_ls_stdout = await crane_ls.stdout()
+        except ExecError as e:
+            if "NAME_UNKNOWN" in e.stderr:
                 return StepResult(self, status=StepStatus.SUCCESS, stdout=f"The docker repository {docker_repository} does not exist.")
             else:
-                return StepResult(self, status=StepStatus.FAILURE, stderr=crane_ls_stderr, stdout=crane_ls_stdout)
+                return StepResult(self, status=StepStatus.FAILURE, stderr=e.stderr, stdout=e.stdout)
```

Comment:
This is another good example of replacing `if exit_code != 0`.

Notice how in a success condition, only stdout is needed here, that's why all we need is a simple `await crane_ls.stdout()` inside `try` instead of using the utils above.